File: SoftUniCourses/JavaScript/VSCode/JavaScript/whileExe/cake.js
function cake(input) {

    let length = Number(input[0]);
    let width = Number(input[1]);

    let sizeCake = length * width;

    let i = 2;
    let command = input[i];
    let piecesNum = 0;

    while (command !== "STOP") {
        let partsCake = Number(command);
        piecesNum = piecesNum + partsCake;

        if (piecesNum > sizeCake) {
            console.log(`No more cake left! You need ${piecesNum - sizeCake} pieces more.`);
            break;
        }
        i++;
        command = input[i];
    }
    if (command === "STOP") {
        if (piecesNum <= sizeCake) {
            console.log(`${sizeCake - piecesNum} pieces are left.`);

        }
    }
}
cake(["10", "10", "20", "20", "20", "20", "21"]);

File: SoftUniCourses/SoftUniBasics/whileloops/Cake.java
package whileloops;

import java.util.Scanner;

public class Cake {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int width =Integer.parseInt(scanner.nextLine());
        int length =Integer.parseInt(scanner.nextLine());

        int countPieces = width * length;//лице на торта-общо пар1етата

        String command = scanner.nextLine();
        while (!command.equals("STOP")){
            int takePieces = Integer.parseInt(command);
            countPieces -= takePieces;
            if (countPieces < 0){
                System.out.printf("No more cake left! You need %d pieces more", Math.abs(countPieces));
                break;
            }
          command = scanner.nextLine();

        }                           //край w
         if (command.equals("STOP")){
             System.out.printf("%d pieces are left.", countPieces);
         }

    }
}
